gcp_variant_transforms/libs/annotation/vep/vep_runner.py

```diff
@@ -0,0 +1,464 @@
+        many and what type of workers to use, where to run, etc.
+    """
+    self._pipeline_service = pipeline_service
+    self._vep_image_uri = vep_image_uri
+    self._vep_cache_path = vep_cache_path
+    self._vep_num_fork = vep_num_fork
+    self._input_pattern = input_pattern
+    self._output_dir = output_dir
+    self._vep_info_field = vep_info_field
+    self._process_pipeline_args(pipeline_args)
+    # _current_operations holds the list of operation names currently running.
+    self._running_operation_ids = []  # type: List[str]
+
+  def get_output_pattern(self):
+    # type: () -> str
+    return (self._format_dir_path(self._output_dir) +
+            _GCS_RECURSIVE_WILDCARD + _VEP_OUTPUT_SUFFIX)
+
+  def _get_api_request_fixed_parts(self):
+    # type: () -> Dict
+    """Returns the part of API request that is fixed between actions.
+
+    This includes setting up VEP cache, virtual machine setup, etc. The variant
+    parts are the `commands` for processing each file and should be added before
+    sending the API request.
+    """
+    return {
+        _API_PIPELINE: {
+            _API_ACTIONS: [
+                self._make_action('mkdir', '-p', '/mnt/vep/vep_cache'),
+                self._make_action('gsutil', '-q', 'cp', self._vep_cache_path,
+                                  '/mnt/vep/vep_cache/')
+            ],
+            'environment': {
+                'VEP_CACHE': '/mnt/vep/vep_cache/{}'.format(
+                    _get_base_name(self._vep_cache_path)),
+                'NUM_FORKS': str(self._vep_num_fork),
+                'VCF_INFO_FILED': self._vep_info_field,
+                # TODO(bashir2): Decide how to do proper reference validation,
+                # the following --check_ref just drops variants that have
+                # wrong REF. If there are too many of them, it indicates that
+                # VEP database for a wrong reference sequence is being used
+                # and this has to caught and communicated to the user.
+                'OTHER_VEP_OPTS':
+                    '--everything --check_ref --allow_non_variant',
+            },
+            'resources': {
+                'projectId': self._project,
+                'virtualMachine': {
+                    'disks': [
+                        {
+                            'name': 'vep',
+                            'sizeGb': _MINIMUM_DISK_SIZE_GB
+                        }
+                    ],
+                    'machineType': self._machine_type,
+                    # TODO(bashir2): Add the option of using preemptible
+                    # machines and the retry functionality.
+                    'preemptible': False,
+                    'serviceAccount': {
+                        'scopes': [
+                            'https://www.googleapis.com/auth/'
+                            'devstorage.read_write']
+                    }
+                },
+                'regions': [
```

Comment:
I made this particular suggestion and one more in `_make_action` below. Not sure about the dsub link you shared because it is in `is_pipelines_operation` function (maybe you meant to share a permanent link?) but in any case, there are probably some more ways to make these more compact that may or may not hinder readability. I am fine the way it is now.